### PR TITLE
Add workflows, samples, tests, and formatting checks

### PR DESCRIPTION
**Pull request by Shane32**
Overview of changes:
- Run `dotnet format` and apply all recommended changes
- Added workflows based on those used in graphql-dotnet/server, and removed old workflow
  - Test code when creating/updating a PR
  - Check formatting when creating/updating a PR
  - Fail if PR title contains '[WIP]'
  - Build code and upload nuget preview package to GitHub Packages when merging to master/develop
  - Publish to nuget.org when creating a release
- Added nuget secret to repository secrets
- Added readme and workflows to solution
- Added sample projects to solution, so they build during tests (todo: add tests)
- Remove outdated dependencies on GraphQL.SystemReactive and replace with System.Reactive
- Update subscription sample to use latest version of GraphQL.Server; update code and references as necessary
- Update SimpleWebApp to support Newtonsoft.Json (add `AllowSynchronousIO = true`)
- Add Playground UI package to SimpleWebApp and DataLoaderWithEFCore so they can be tested (neither works however)
- Update all GraphQL dependencies to 7.2.0
- Remove unnecessary dependencies from GraphQL.Conventions
- Update GraphQL.Conventions version to 7.1.0
- Added test project for SubscriptionExample, which verifies both POST and WebSocket subscription operations
- Add `schema.Initialize()` to 3 of 4 SchemaConstructorTests

No functional changes to GraphQL.Conventions were made in this PR, although some unnecessary/obsolete dependencies were removed.  Users may need to add a reference to System.Reactive 5.0.0.

Appveyor integrations may be removed with the merging of this PR.

Subsequent PRs might include these changes:
- Add Directory.build.props
- Add Tests.props
- Bump to .NET 7
- Add editorconfig
- Update csproj properties
- Remove global.json

Possible future changes:
- Remove or deprecate 'server'-type functionality; instead rely on the server project
- Add configuration extensions to `IGraphQLBuilder`
- Remove GraphQL.DataLoader dependency (maintaining support for data loaders)
- Replace `IDependencyInjector` with `IServiceProvider`

### CODE REVIEW

**Comment by tlil** [comment]
Thanks for this @Shane32 - will review properly on the weekend, but at first glance this looks great.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup .NET SDK
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0.x
```
Comment (sungam3r) [inline .github/workflows/build.yml:23]: Why not 7 ?

File: Conventions.sln [hunk 2]
```diff
@@ -1,10 +1,44 @@
...
+		.github\workflows\build.yml = .github\workflows\build.yml
+		.github\workflows\formatting.yml = .github\workflows\formatting.yml
+		.github\workflows\publish.yml = .github\workflows\publish.yml
+		.github\workflows\test.yml = .github\workflows\test.yml
+		.github\workflows\wipcheck.yml = .github\workflows\wipcheck.yml
+	EndProjectSection
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".Solution Items", ".Solution Items", "{5A8618B7-7299-4507-A665-CC9CEE8B4674}"
+	ProjectSection(SolutionItems) = preProject
+		global.json = global.json
```
Comment (sungam3r) [inline Conventions.sln:34]: I doubt about this file. In other OSS projects that have it I should always touch/fix it to build code locally.

File: samples/DataLoaderWithEFCore/DataLoaderWithEFCore/DataLoaderWithEFCore.csproj [hunk 3]
```diff
@@ -15,8 +15,9 @@
   <ItemGroup>
     <PackageReference Include="AutoMapper" Version="11.0.1" />
     <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="11.0.0" />
-    <PackageReference Include="GraphQL" Version="5.3.0" />
-    <PackageReference Include="GraphQL.DataLoader" Version="5.3.0" />
+    <PackageReference Include="GraphQL" Version="7.2.0" />
```
Comment (sungam3r) [inline samples/DataLoaderWithEFCore/DataLoaderWithEFCore/DataLoaderWithEFCore.csproj]:
```suggestion
```

File: samples/SimpleWebApp/Program.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ public class Program
         public static void Main(string[] args)
         {
             var host = new WebHostBuilder()
-                .UseKestrel()
+                .UseKestrel(options => options.AllowSynchronousIO = true)
```
Comment (sungam3r) [inline samples/SimpleWebApp/Program.cs]:
```suggestion
                .UseKestrel(options => options.AllowSynchronousIO = true) // for Newtonsoft.Json support
```

File: samples/SimpleWebApp/SimpleWebApp.csproj [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
```
Comment (sungam3r) [inline samples/SimpleWebApp/SimpleWebApp.csproj:4]: I would change all samples to net7

File: test/GraphQL.Conventions.Tests/GraphQL.Conventions.Tests.csproj [hunk 6]
```diff
@@ -12,11 +12,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="GraphQL" Version="7.0.0" />
-    <PackageReference Include="GraphQL.DataLoader" Version="7.0.0" />
-    <PackageReference Include="GraphQL.SystemReactive" Version="4.8.0" />
+    <PackageReference Include="GraphQL" Version="7.2.0" />
```
Comment (sungam3r) [inline test/GraphQL.Conventions.Tests/GraphQL.Conventions.Tests.csproj]:
```suggestion
```

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj [hunk 7]
```diff
@@ -2,8 +2,8 @@
 
   <PropertyGroup>
     <Description>GraphQL Conventions for .NET</Description>
-    <VersionPrefix>7.0.0</VersionPrefix>
-    <PackageVersion>7.0.0</PackageVersion>
+    <VersionPrefix>7.1.0</VersionPrefix>
+    <PackageVersion>7.1.0</PackageVersion>
     <Authors>Tommy Lillehagen</Authors>
     <TargetFrameworks>net6.0;netstandard2.0</TargetFrameworks>
     <LangVersion>9.0</LangVersion>
```
Comment (sungam3r) [inline src/GraphQL.Conventions/GraphQL.Conventions.csproj:9]: I would change to latest in dirbuildprops + also remove DepsFileGenerationMode, AssemblyName, PackageId, GeneratePackageOnBuild, both DefineConstants + change GenerateDocumentationFile to true and DebugType to embedded.

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj [hunk 8]
```diff
@@ -29,14 +27,9 @@
...
   <ItemGroup>
-    <PackageReference Include="GraphQL" Version="7.0.0" />
-    <PackageReference Include="GraphQL.DataLoader" Version="7.0.0" />
-    <PackageReference Include="GraphQL.NewtonsoftJson" Version="7.0.0" />
-    <PackageReference Include="GraphQL.SystemReactive" Version="4.8.0" />
-    <PackageReference Include="System.Reflection" Version="4.3.0" />
-    <PackageReference Include="System.Runtime" Version="4.3.1" />
-    <PackageReference Include="System.Reflection.Extensions" Version="4.3.0" />
-    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.7.0" />
+    <PackageReference Include="GraphQL" Version="7.2.0" />
```
Comment (sungam3r) [inline src/GraphQL.Conventions/GraphQL.Conventions.csproj]:
```suggestion
```

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj [hunk 9]
```diff
@@ -29,14 +27,9 @@
...
+    <PackageReference Include="GraphQL.DataLoader" Version="7.2.0" />
+    <PackageReference Include="GraphQL.NewtonsoftJson" Version="7.2.0" />
```
Comment (sungam3r) [inline src/GraphQL.Conventions/GraphQL.Conventions.csproj:31]: 🤔 strongly coupled with NSJ?

File: test/SubscriptionExample.Tests/EndToEndTests.cs [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+                  ""text"": ""I like turtles""
+                }
+              }
+            }";
+
+        var query5 = query1;
+
+        var expected5 = @"
+            {
+              ""data"": {
```
Comment (sungam3r) [inline test/SubscriptionExample.Tests/EndToEndTests.cs:84]: good candidates for raw string literals

File: test/SubscriptionExample.Tests/SubscriptionExample.Tests.csproj [hunk 11]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<PropertyGroup>
+		<TargetFramework>net6.0</TargetFramework>
```
Comment (sungam3r) [inline test/SubscriptionExample.Tests/SubscriptionExample.Tests.csproj:4]: Whitespaces instead of tabs.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Shane32) [reply .github/workflows/build.yml:23]: This PR does not bump any versions to .NET 7

File: Conventions.sln (hunk 2 above)
Comment (Shane32) [reply Conventions.sln:34]: I do not know what it is there for; I would delete it.  I would propose the change in a separate PR so @tlil can review.

File: samples/SimpleWebApp/SimpleWebApp.csproj (hunk 5 above)
Comment (Shane32) [reply samples/SimpleWebApp/SimpleWebApp.csproj:4]: Personally, I would not, as .NET 7 is not a LTS version, but it doesn't matter so much.

However, bumping to .NET 7 should be done in its own PR regardless.

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj (hunk 7 above)
Comment (Shane32) [reply src/GraphQL.Conventions/GraphQL.Conventions.csproj:9]: Those changes should be made in a separate PR.

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj (hunk 9 above)
Comment (Shane32) [reply src/GraphQL.Conventions/GraphQL.Conventions.csproj:31]: The Conventions project has its own server; deserializing requests and serializing responses.  It is not within the scope of this PR to remove or modify such behavior.

File: test/SubscriptionExample.Tests/EndToEndTests.cs (hunk 10 above)
Comment (Shane32) [reply test/SubscriptionExample.Tests/EndToEndTests.cs:84]: True, but until `LangVersion` is bumped I could not do so.  Another PR.

File: test/SubscriptionExample.Tests/SubscriptionExample.Tests.csproj (hunk 11 above)
Comment (Shane32) [reply test/SubscriptionExample.Tests/SubscriptionExample.Tests.csproj:4]: Auto-format due to missing .editorconfig.  Another PR.

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj (hunk 7 above)
Comment (Shane32) [reply src/GraphQL.Conventions/GraphQL.Conventions.csproj:9]: At least, I'd prefer to make those changes in a separate PR where they can be independently reviewed.  This PR really is just trying to get the basic workflows in place to switch 1:1 to GitHub Actions from Appveyor, and to get the test projects usable -- which probably should have been two PRs.

Comment (sungam3r) [reply src/GraphQL.Conventions/GraphQL.Conventions.csproj:9]: Ok

File: src/GraphQL.Conventions/GraphQL.Conventions.csproj (hunk 9 above)
Comment (sungam3r) [reply src/GraphQL.Conventions/GraphQL.Conventions.csproj:31]: Just a question.

**Review by sungam3r — APPROVED** [review]

**Review by tlil — APPROVED** [review]

**Comment by tlil** [comment]
Merged @Shane32  - thanks.

Only hiccup I came across was when tagging a release version with the new GH action scripts:
CleanShot 2022-12-10 at 18 43 51

See: https://github.com/graphql-dotnet/conventions/actions/runs/3665473254/jobs/6196568538#step:6:12

I've disabled the AppVeyor webhook by the way.

**Comment by Shane32** [comment]
It’s designed for a tag without v - e.g. “7.1.0” not “v7.1.0”

I can change the script but then of course it would be different than the other repos. Lmk

**Comment by Shane32** [comment]
FYI, my changes didn’t fix #237

**Comment by tlil** [comment]
> It’s designed for a tag without v - e.g. “7.1.0” not “v7.1.0”
> 
> I can change the script but then of course it would be different than the other repos. Lmk

Gotcha. Suspected that was the case. Can change convention in this repo #pun-intended :) So no need to change the script. Thanks again

**Comment by tlil** [comment]
> FYI, my changes didn’t fix #237 

Will look at this today. 👍

**Pull request merged**